7.application_configuration.md

```diff
@@ -159,239 +147,30 @@ metadata:
     version: v1.0.0
     description: "Description of this deployment"
 spec:
-  variables:
-    - name: VAR_NAME
-      value: SUPPLIED_VALUE
-  scopes:
-    - name: core.oam.dev/v1alpha1.Network
-      parameterValues:
-        - name: PARAM_NAME
-          value: SUPPLIED_VALUE
   components:
     - componentName: my-web-app-component
-      instanceName: my-app-frontent
       parameterValues:
         - name: PARAMETER_NAME
           value: SUPPLIED_VALUE
         - name: ANOTHER_PARAMETER
-          value: "[fromVariable(VAR_NAME)]"
+          value: "AnotherValue"
       traits:
-        - name: ingress
+        - name: manualscaler.core.oam.dev
+          version: v1
           properties:
-            DATA: "[fromVariable(VAR_NAME)]"
-
+            replicaCount: 3
+      scopes:
+        - scopeRef:
+            apiVersion: core.oam.dev/v1alpha1
+            kind: Network
+            name: example-vpc-network
```

Comment:
If this is a reference to an instance of a scope, why do I need to specify version and kind here? Shouldn't the name of the instance be sufficient?

I think the `Component` section above has it (mostly) right in defining this as just a list of strings, and this example is wrong.

Reply:
We need the `apiVersion` and kind because scopes may be of an arbitrary version and kind (e.g. `NetworkScope` , `HealthScope`, etc). We can't uniquely identify the referenced scope by name alone.